controllers/provider-aws/pkg/infrastructure/actuator.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not render chart", 50, err)
+		return err
+	}
+
+	tf, err := c.getTerraformer(awstypes.TerrformerPurposeInfra, infrastructure.Namespace, infrastructure.Name)
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not create terraformer object", 70, err)
+		return err
+	}
+
+	if err = tf.SetVariablesEnvironment(terraformer.GenerateVariablesEnvironment(providerSecret, map[string]string{
+		"ACCESS_KEY_ID":     awstypes.AccessKeyID,
+		"SECRET_ACCESS_KEY": awstypes.SecretAccessKey,
+	})).InitializeWith(terraformer.DefaultInitializer(c.client,
+		release.FileContent("main.tf"),
+		release.FileContent("variables.tf"),
+		[]byte(release.FileContent("terraform.tfvars")))).Apply(); err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not apply terraform config", 80, err)
+		return err
+	}
+
+	if err = c.injectProviderStateIntoStatus(context.TODO(), tf, infrastructure); err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "failed to update status with the provider state", 90, err)
+		return err
+	}
+
+	return c.injectStatusSuccess(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "successfully reconciled infrastructure")
+}
+
+func (c *actuator) delete(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	c.logger.Info("Destroying Infrastructure")
+
+	tf, err := c.getTerraformer(awstypes.TerrformerPurposeInfra, infrastructure.Namespace, infrastructure.Name)
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeDelete, "could not create terraformer object", 30, err)
+		return err
+	}
+
+	configExists, err := tf.ConfigExists()
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeDelete, "terraform configuration was not found on seed", 60, err)
+		return err
+	}
+
+	if configExists {
+		var (
+			namespace = infrastructure.Spec.SecretRef.Namespace
+			name      = infrastructure.Spec.SecretRef.Name
+			region    = infrastructure.Spec.Region
+		)
+
+		providerSecret := &corev1.Secret{}
+		if err := c.client.Get(ctx, kutil.Key(namespace, name), providerSecret); err != nil {
+			return err
+		}
+
+		awsClient := aws.NewClient(string(providerSecret.Data[awstypes.AccessKeyID]), string(providerSecret.Data[awstypes.SecretAccessKey]), region)
+
+		c.logger.Info("Destroying Kubernetes load balancers and security groups")
+		err := c.destroyKubernetesLoadBalancersAndSecurityGroups(namespace, tf, awsClient)
+		if err != nil {
+			c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeDelete, "failed to destroy kubernetes load balancers and security groups", 80, err)
+		}
+	}
+
```

Comment:
Where is `terraform destroy` called? Why not using the `flow` package as [it is currently implemented](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/awsbotanist/infrastructure.go#L71-L88)?